### PR TITLE
chore(HelperText): update tests

### LINKED ISSUE #9760: Helper text test updates

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #9760

### CODE REVIEW

File: packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx [hunk 1]
```diff
@@ -1,66 +1,75 @@
...
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem>help test text 1</HelperTextItem>
-        <HelperTextItem>help test text 2</HelperTextItem>
-        <HelperTextItem>help test text 3</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders as ul when component = ul', () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx:48]:
```suggestion
test('Renders with element passed to component prop', () => {
```

We should also have a test that checks the top element is a div by default

File: packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx [hunk 2]
```diff
@@ -1,66 +1,75 @@
...
+  render(
+    <HelperText component="ul" aria-label="helper">
+      <HelperTextItem component="li">help test text 1</HelperTextItem>
+      <HelperTextItem component="li">help test text 2</HelperTextItem>
+    </HelperText>
+  );
+  expect(screen.getByLabelText('helper').tagName).toBe('UL');
+  expect(screen.getByText('help test text 1').parentElement?.tagName).toBe('LI');
```
Comment (thatblindgeye) [inline packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx:56]: Let's split this up into separate tests. This test should check that the `ul` tag is there, then for aria-label we can have tests for:

- checking the aria-label isn't rendered if the component is not ul
- checking the aria-label does get rendered when component = ul

We should also add a couple of tests that check the `role` attribute doesn't get applied when component is not ul, and that it does get applied when component = ul

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
We should also have separate files for HelperText and HelperTextItem tests.

For Helper Text we should add tests for:
- id on HelperText is rendered
- a test for "renders without children" like Alert:
    https://github.com/patternfly/patternfly-react/blob/1194a7e9e99e6377906799f2c142c0b843112256/packages/react-core/src/components/Alert/__tests__/Alert.test.tsx#L29-L34
- spreads additional props correctly
- passing isLiveRegion sets the aria-live attribute

For HelperTextItem we should add tests for:
- renders without children like above
- renders with the default class on both the top level element (helperTextItem) and the element the text is rendered in (helperTextItemText)
- renders with passed id
- renders an icon when hasIcon is passed && icon is not passed (nice to have would be testing that there aren't 2 instances of the helperTextItemIcon when both hasIcon and icon are passed)
- the element is rendered with a div by default and renders with whatever is passed to `component`
- `screenReaderText` renders when it and isDynamic is passed

File: packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx [hunk 3]
```diff
@@ -1,66 +1,66 @@
...
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders default classes', () => {
+  render(<HelperText>text</HelperText>);
+  expect(screen.getByText('text')).toHaveClass(styles.helperText);
+});
+
+test('Renders id when id is passed', () => {
+  render(<HelperText id="helper-id">text </HelperText>);
+  expect(screen.getByText('text')).toHaveAttribute('id');
```
Comment (thatblindgeye) [inline packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx:29]:
```suggestion
  expect(screen.getByText('text')).toHaveAttribute('id', 'helper-id');
```

File: packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx [hunk 4]
```diff
@@ -1,66 +1,66 @@
...
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem icon={<CheckIcon />}>help test text</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders aria-live when isLiveRegion is passed', () => {
+  render(<HelperText isLiveRegion>text </HelperText>);
+  expect(screen.getByText('text')).toHaveAttribute('aria-live');
```
Comment (thatblindgeye) [inline packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx:34]:
```suggestion
  expect(screen.getByText('text')).toHaveAttribute('aria-live', 'polite');
```

Could we also do the inverse, checking aria-live doesn't render by default?

File: packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx [hunk 5]
```diff
@@ -1,66 +1,66 @@
...
-  test('dynamic helper text renders successfully', () => {
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem isDynamic>help test text</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Spreads additional props when passed', () => {
+  render(<HelperText>text </HelperText>);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx:38]: Just need to pass another prop in here to test for

File: packages/react-core/src/components/HelperText/__tests__/HelperTextItem.test.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      help test text
+    </HelperTextItem>
+  );
+  expect(screen.getByText('test').parentElement).toHaveClass(styles.helperTextItemIcon);
+});
+
+test('Renders dynamic helper text', () => {
+  render(<HelperTextItem isDynamic>help test text</HelperTextItem>);
+  expect(screen.getByText('help test text').parentElement).toHaveClass(styles.modifiers.dynamic);
+  expect(screen.getByText('help test text').querySelector('span')).toHaveClass('pf-v5-screen-reader');
```
Comment (thatblindgeye) [inline packages/react-core/src/components/HelperText/__tests__/HelperTextItem.test.tsx:84]: Can we add another test to check that custom screenReaderText works

File: packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx [hunk 7]
```diff
@@ -1,66 +1,66 @@
...
+  expect(screen.getByText('text')).toBeInTheDocument();
+});
+
+test('Renders custom className', () => {
+  render(<HelperText className="custom">text </HelperText>);
+  expect(screen.getByText('text')).toHaveClass('custom');
+});
+
+test('Renders with element passed to component prop', () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx:47]: Just need a test that's the inverse to check it renders with a div by default

**Review by thatblindgeye — COMMENTED** [review]
Thanks for the updates! Just a few more quick ones below

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by thatblindgeye — APPROVED** [review]

**Pull request merged**